src/mscorlib/src/System/Reflection/RuntimeMethodInfo.cs

```diff
@@ -468,15 +468,22 @@ private void ThrowNoInvokeException()
         [DebuggerStepThroughAttribute]
         [Diagnostics.DebuggerHidden]
         public override Object Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
+        {
+            return InvokeInternal(obj, invokeAttr, binder, parameters, culture, false, null);
+        }
+
+        [DebuggerStepThroughAttribute]
+        [Diagnostics.DebuggerHidden]
+        internal Object InvokeInternal(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture, bool propertyGetterCalledThroughSetValue, Object refReturnPropertySetValue)
```

Comment:
This extra layer and arguments are going to regress reflection Invoke performance up to 5%:

```csharp
public class Test
{
    public static void DoNothing() { }

    static void Main() {
        MethodInfo mi = typeof(Test).GetMethod("DoNothing");

        int start = Environment.TickCount;
        for (int i = 0; i < 100000000; i++) mi.Invoke(null, null);
        int end = Environment.TickCount;
        Console.WriteLine((end-start).ToString());
    }
}
```